src/components/Identity/injectEnsureRequestHasIdentity.js

```diff
@@ -18,38 +18,72 @@ export default ({
   awaitIdentityCookie,
   logger
 }) => {
-  let identityCookiePromise;
+  let obtainedIdentityPromise;
+
+  const allowRequestToGoWithoutIdentity = payload => {
+    setDomainForInitialIdentityPayload(payload);
+    return addLegacyEcidToPayload(payload);
+  };
+
   /**
-   * Ensures that if no identity cookie exists, we only let one request be
-   * sent without an identity until its response returns. In the meantime,
+   * Ensures that if no identity cookie exists, we only let one request at a
+   * time without an identity until its response returns. In the meantime,
    * we queue all other requests, otherwise the requests could result in
-   * multiple ECIDs being minted for the user. Once the response to the first
-   * request returns, we can let the queued requests be sent, since they
-   * will have the newly minted ECID that was returned on the first response.
+   * multiple ECIDs being minted for the user. Once we get an identity
+   * cookie, we can let the queued requests be sent all at once, since they
+   * will have the newly minted ECID.
+   *
+   * Konductor should make every effort to return an identity, but in
+   * certain scenarios it may not. For example, in cases where the
+   * request does not match what Konductor is expecting (ie 400s).
+   * In cases where Konductor does not set an identity, there should be
+   * no events recorded so we don't need to worry about multiple ECIDs
+   * being minted for each user.
+   *
+   * If we only allowed one request through without an identity, a single
```

Comment:
Maybe reword this a bit to say something like: the reason we allow for multiple sequential requests to be sent without an identity is to prevent a single malformed ...